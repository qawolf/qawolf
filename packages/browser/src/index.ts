// public API
export { BrowserContext, register } from "./ContextManager";
export { Page } from "./page/PageManager";
// TODO move to jest-playwright
export { launch, LaunchResult } from "./launch";

// internal API
export * from "./actions";
export * from "./keyboard";
export { ContextManager } from "./ContextManager";
export { PageManager } from "./page/PageManager";
