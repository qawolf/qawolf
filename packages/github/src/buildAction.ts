import { readFileSync } from "fs-extra";
import { compile } from "handlebars";
import { resolve } from "path";

const acitonTemplate = compile(
  readFileSync(resolve(__dirname, "../static/action.hbs"), "utf8")
);

export const buildAction = (): string => {
  return acitonTemplate({});
};
