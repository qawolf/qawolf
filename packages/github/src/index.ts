import Octokit from "@octokit/rest";

const octokit = new Octokit({
  auth: "d9e1b19607b2b07ed5c6f2e78fa5fb1bd3cabf26"
});

const main = async () => {
  const res = await octokit.checks.create({
    owner: "qawolf",
    repo: "internal",
    name: "QA Wolf",
    head_sha: "9c52a7a6f2090c58bcb23ae066ce31920789f71c"
  });

  console.log("RESPONSE", res);
};

main();
