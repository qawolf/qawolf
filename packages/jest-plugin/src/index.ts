// setup fail-fast since we do not want to continue when a step fails
// hopefully jest-circus implements this behavior, but was not the case when I tested jest 24.9.0
// https://github.com/facebook/jest/issues/2867#issuecomment-476647864
// https://github.com/facebook/jest/issues/6527#issuecomment-506278251

type Callback = (name: string) => void;

const j = jasmine as any;

// require since there are no types
const failFast = require("@qawolf/jasmine-fail-fast");
const jasmineEnv = j.getEnv();
jasmineEnv.addReporter(failFast.init());
