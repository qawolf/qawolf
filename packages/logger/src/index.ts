import { Logger } from "./Logger";

export const browserLogger = new Logger("browser");

export const logger = new Logger("server");
