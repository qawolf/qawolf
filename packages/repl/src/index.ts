import { Reporter } from "./Reporter";

// export for types to be happy
export { repl, ReplWithContext } from "./ReplWithContext";
export { CONTEXT } from "./Context";

import { repl } from "./ReplWithContext";
import { CONTEXT } from "./Context";

// export with module.exports for Jest to be happy
const mainExport = Reporter as any;
mainExport.CONTEXT = CONTEXT;
mainExport.repl = repl;
module.exports = mainExport;
