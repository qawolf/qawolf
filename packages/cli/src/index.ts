#!/usr/bin/env node

import { CONFIG } from "@qawolf/config";
import { logger } from "@qawolf/logger";
import program from "commander";
import { snakeCase } from "lodash";
import { githubAction } from "./githubAction";
import { record } from "./record";
import { test } from "./test";
import { parseUrl } from "./utils";

let recordCommand = program
  .command("record <url> [name]")
  .description("record a workflow and create a test");

if (CONFIG.development) {
  recordCommand = recordCommand.option(
    "-e, --events",
    "save events (for debugging)"
  );
}

recordCommand.action(async (urlArgument, optionalName, cmd) => {
  const url = parseUrl(urlArgument);
  logger.verbose(`record url "${url.href}"`);

  const name = snakeCase(optionalName || url.hostname!);
  await record(url, name, cmd.events);
});

program
  .command("test [name]")
  .description("run a test")
  .action(async optionalName => {
    await test(optionalName);
  });

program
  .command("github")
  .description("Generate workflow file for GitHub Actions")
  .action(async () => {
    await githubAction();
  });

program.allowUnknownOption(false);

program.parse(process.argv);

if (!process.argv.slice(2).length) {
  console.log("\n");
  program.outputHelp();
}
