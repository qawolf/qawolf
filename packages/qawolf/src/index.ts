#!/usr/bin/env node
import updateNotifier from "update-notifier";

const pkg = require("../package.json");

const isCLI = !module.parent;
if (isCLI) {
  updateNotifier({ pkg }).notify();

  require("@qawolf/cli");
}

// export public API
export { BrowserContext, launch, Page } from "@qawolf/browser";

export { repl } from "@qawolf/repl";

export { sleep, waitFor, waitUntil } from "@qawolf/web";

// make repl a global
import { repl } from "@qawolf/repl";
(global as any).repl = repl;
