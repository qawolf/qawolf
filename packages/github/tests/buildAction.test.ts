import { buildAction } from "../src/buildAction";

describe("buildAction", () => {
  it("returns action template", () => {
    const action = buildAction();

    expect(action).toMatchSnapshot();
  });
});
