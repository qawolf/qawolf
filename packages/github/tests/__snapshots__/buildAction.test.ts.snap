// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildAction returns action template 1`] = `
"name: qawolf
    on:
        push:
            # test every branch
            # edit below if you only want certain branches tested
            branches: '*'
        # schedule:
            ## test on schedule using cron syntax
            # - cron \\"0 * * * *\\"
jobs:
    test:
        runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
        with:
            # copy only latest version of repo
            fetch-depth: 1
      - name: qawolf test
        uses: qawolf/qawolf@master
        env:
            # parameterize test values with env variables
            # edit QAW_URL to have all tests go to different url than recording url
            # QAW_URL: www.my_staging_url.com
            # edit these env variables to change url for specific test (example: QAW_LOGIN_URL)
            # QAW_TESTNAME_URL: www.my_login_test_url.com
            # edit these env variables to change input value for test step (example: QAW_LOGIN_1)
            # QAW_TESTNAME_STEP: \\"my new input value\\"
            # pass a secret as env variable (create secret in GitHub repo settings)
            # QAW_TESTNAME_STEP: \${{ secrets.PASSWORD }}
      - name: Upload Artifacts
        # include test artifacts
        # edit below to only include artifacts in certain scenarios
        if: always()
        uses: actions/upload-artifact@master
        with:
            name: qawolf
            path: './artifacts'"
`;
